### DISCUSSION TITLE
Error "Failed to parse COSE_Sign1 outer array" when running make demo on banking-app demo using CCF

### DISCUSSION THREAD

**Discussion by aframicrosoft** [microsoft/CCF#6397] [Q&A]
Hi everyone,

I am trying to get the [banking-app demo](https://github.com/microsoft/ccf-app-samples/tree/main/banking-app) running on Visual Studio Code (either locally or on codespaces). However, I've hit a roadblock with the following error message when executing `make demo`:

```
+ cat
+ cp ../../vote/vote_accept.json ./
+++ date -uIs
++ curl -s https://127.0.0.1:8000/gov/proposals --cacert service_cert.pem --data-binary @- -H 'content-type: application/cose'
++ ccf_cose_sign1 --ccf-gov-msg-type proposal --ccf-gov-msg-created_at 2024-07-25T21:02:11+00:00 --signing-key member0_privk.pem --signing-cert member0_cert.pem --content set_user0.json
../../scripts/demo_governance.sh: line 50: ccf_cose_sign1: command not found
+ proposal0_out='{"error":{"code":"InternalError","message":"Failed to parse COSE_Sign1 outer array"}}'
```

Here's a summary of what I've done so far:

1. Installed Python and CCF on my local machine.
2. Followed the setup instructions for the banking-app demo.

Despite these steps, I keep encountering the error mentioned above.

Questions:

- Has anyone faced a similar issue and found a solution?
- Could there be additional dependencies or configurations I'm missing?
- Any specific versions of Python or other tools recommended for running this demo?

Any help or guidance would be greatly appreciated!

Thanks in advance!

**Comment by achamayou** [comment]
Hi @aframicrosoft, the ccf_cose_sign1 CLI tool is part of the ccf Python package (https://pypi.org/project/ccf/), installing that should resolve your problem (`pip install ccf`).

**Comment by eddyashton** [comment]
Thanks for raising. Fix to the demo script is incoming: https://github.com/microsoft/ccf-app-samples/pull/254

As @achamayou says, you can also manually install the necessary packages - run `pip install ccf` on the second terminal before `make demo`.
